<launch>

#### start the sick node #######################################
<node pkg="sicktoolbox_wrapper" type="sicklms"
name="sicklms" output="screen">

	<param name="port" value="/dev/ttyUSB0" />

</node>

<!--
#### start sensor drivers ######################################
<node pkg="ION" type="sb_gyro_accelerometer.py" name="sb_gyro_accelerometer" output="screen"/>
-->

#### publish initial base_link -> laser transform ##############
<node pkg="tf" type="static_transform_publisher" name="base_link_to_laser" args=".254 0 0 0 0 0 /base_link /laser 40"/>

#### start the laser scan_matcher ##############################
<node pkg="laser_scan_matcher" type="laser_scan_matcher_node"
name="laser_scan_matcher_node" output="screen">

    <param name="fixed_frame" value = "odom"/>
    <param name="use_alpha_beta" value="true"/>
    <param name="use_odom" value="false"/>
    <param name="use_imu" value="false"/>
    <param name="max_iterations" value="10"/>
    <param name="publish_pose" value="true"/>
    <param name="publish_tf" value="true"/>
    <param name="use_vel" value="false"/>

</node>

#### start slam_gmapping #######################################
<node pkg = "gmapping" type="slam_gmapping"
name="slam_gmapping" output="screen">

    <param name="map_update_interval" value="0.5"/>
    <param name="maxUrange" value="4.0"/>
    <param name="linear_update" value="0.2"/>
    <param name="angularUpdate" value="0.2"/>
    <param name="delta" value="0.05"/>
    <param name="occ_thresh" value="0.1"/>

</node>

#### start waypoint_creator ####################################
<node pkg="ION" type="sb_waypoint_creator"
name="sb_waypoint_creator" output="screen"/>

#### start move_straight_line ##################################
<node pkg="ION" type="move_straight_line"
name="move_straight_line" output="screen">
    <param name="explicit_turn_threshold" value="0.1"/>
    <param name="stop_threshold" value="0.2"/>
    <param name="move_speed" value="2.0"/>
</launch>
